tests/test_type/test_tuple.py

```diff
@@ -408,3 +408,14 @@ class tuple_key_ordering(m.Circuit):
 """
 
     assert has_error(caplog, msg)
+
+
+def test_tuple_same_key_value():
+    T0 = m.AnonProduct[{"x": m.Bit, "y": m.Bits[8]}]
+    T1 = m.AnonProduct[{"y": m.Bits[8], "x": m.Bit}]
+
+    class tuple_same_key_value(m.Circuit):
+        io = m.IO(I=m.In(T0), O=m.Out(T1))
+        io.O.x @= io.I.x
+        io.O.y @= io.I.y
+        assert isinstance(io.O.value(), m.Out(T1)), type(io.O.value())
```

Comment:
we can probably do all this without an enclosing circuit right?

```suggestion
    I = m.Out(T0)
    O = m.In(T1)
    O.x @= I.x
    O.y @= I.y
    assert isinstance(O.value(), m.Out(T1)), type(O.value())
```